File: app/src/main/java/com/github/willjgriff/ethereumwallet/ui/settings/adapter/ChangeAddressItemViewHolder.java
package com.github.willjgriff.ethereumwallet.ui.settings.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.github.willjgriff.ethereumwallet.ethereum.address.model.DomainAddress;
import com.github.willjgriff.ethereumwallet.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by Will on 28/02/2017.
 */

public class ChangeAddressItemViewHolder extends RecyclerView.ViewHolder {

	@BindView(R.id.view_change_address_item_address)
	TextView mAddress;

	private ChangeAddressItemListener mChangeAddressItemListener;

	public ChangeAddressItemViewHolder(View itemView, ChangeAddressItemListener changeAddressItemListener) {
		super(itemView);
		ButterKnife.bind(this, itemView);
		mChangeAddressItemListener = changeAddressItemListener;
	}

	public void bind(DomainAddress domainAddress) {
		mAddress.setText(domainAddress.getHex());
		itemView.setOnClickListener(view -> mChangeAddressItemListener.addressItemClicked(domainAddress));
	}

	public interface ChangeAddressItemListener {
		void addressItemClicked(DomainAddress domainAddress);
	}
}

File: app/src/main/kotlin/com/github/willjgriff/ethereumwallet/ui/screens/settings/list/ChangeAddressItemViewHolder.kt
package com.github.willjgriff.ethereumwallet.ui.screens.settings.list

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView
import com.github.willjgriff.ethereumwallet.ethereum.address.model.DomainAddress
import kotlinx.android.synthetic.main.view_change_address_item.view.*

/**
 * Created by williamgriffiths on 04/05/2017.
 */
class ChangeAddressItemViewHolder(itemView: View,
                                  private val changeAddressItemListener: ChangeAddressItemViewHolder.ChangeAddressItemListener)
    : RecyclerView.ViewHolder(itemView) {

    private var address: TextView = itemView.view_change_address_item_address

    fun bind(domainAddress: DomainAddress) {
        address.text = domainAddress.hex
        itemView.setOnClickListener { _ -> changeAddressItemListener.addressItemClicked(domainAddress) }
    }

    interface ChangeAddressItemListener {
        fun addressItemClicked(domainAddress: DomainAddress)
    }
}